tee-worker/app-libs/stf/src/trusted_call.rs

```diff
@@ -326,9 +338,26 @@ impl TrustedCallVerification for TrustedCallSigned {
 		payload.append(&mut mrenclave.encode());
 		payload.append(&mut shard.encode());
 
-		// make it backwards compatible for now - will deprecate the old way later
-		self.signature.verify(&blake2_256(&payload), self.call.sender_identity())
-			|| self.signature.verify(&payload, self.call.sender_identity())
+		// The signature should be valid in either case:
+		// 1. payload
+		// 2. blake2_256(payload)
+		// 3. Signature Prefix + payload
+		// 4. Signature Prefix + blake2_256payload
+		//
+		// @TODO P-639: Remove 1 and 3.
+
+		let hashed = blake2_256(&payload);
+
+		let prettified_msg_raw = self.call.signature_message_prefix() + &hex_encode(&payload);
+		let prettified_msg_raw = prettified_msg_raw.as_bytes();
+
+		let prettified_msg_hash = self.call.signature_message_prefix() + &hex_encode(&hashed);
+		let prettified_msg_hash = prettified_msg_hash.as_bytes();
+
+		self.signature.verify(&payload, self.call.sender_identity())
```

Comment:
I'm thinking about having an enum representing which type of signature it is instead of naively verifying them one by one.

We could have:
`Raw`
`Prettified`
`CustomPrettified` (containing client defined prefix ?)

Reply:
Yes this is better than trying to verify all cases - I'd suggest doing it in a separate PR though as it would be a breaking change to client.

Additionally, we are going to remove 1 and 3 so we only need to verify twice at maximum. If we know IDHub (the only client for now) is using prettified hash, we could verify this case first to slightly improve the processing time :)

Reply:
> We could have:
> `Raw`
> `Prettified`

If there is nothing against removing 1 and 3, only Prettified version is needed. I suggest moving this discussion to https://linear.app/litentry/issue/P-639/clean-up-remove-raw-payload-signature-verification

> we could verify this case first to slightly improve the processing time

Good call – will do. And yes, IDHub is already using prettified hashes only.